src/Controller/CardController.php

```diff
@@ -66,12 +66,38 @@ public function edit(Nomenclature $nomenclature, Request $request) {
             $currentUser= $this->getUser();
             $nomenclature->setCreatedBy($currentUser);
 
+            $hasDescription = 'No';
+            $hasDescriptionWithGaps = 'No';
+            
             // set card language with the same language as the nomenclature
             $nomLang = $nomenclature->getLanguage();
             foreach ($nomenclature->getCards() as $card){
                 $card->setLanguage($nomLang);
+
+                // test the descripition tag
+                if($card->getDescription() != '') {
+                  if($hasDescription == 'No')
+                    $hasDescription = 'Yes';
+                } else {
+                  if($hasDescription == 'Yes')
+                    $hasDescription = 'Partial';
+                }
+
+                // test the descripition with gaps tag
+                if($card->getDescriptionWithGaps() != '') {
+                  if($hasDescriptionWithGaps == 'No')
+                    $hasDescriptionWithGaps = 'Yes';
+                  } else {
+                    if($hasDescriptionWithGaps == 'Yes')
+                  $hasDescriptionWithGaps = 'Partial';
+                }
```

Comment:
I have an issue with this algorithm...

If first card is empty ('') and all other are provided then it end with a 'Yes' and we wait for a 'Partial'

I think we must count the number of card, count the number of card with empty description and do the math 😄